sliding_window_block.py

```diff
@@ -33,35 +33,35 @@ class SlidingWindow(GroupBy, Block):
 
     """TODO
     - [x] Window Expriation
-    - [ ] Implement Group
+    - [x] Implement Group
     - [ ] Implement Persistence
-    - [ ] Use Signal Expiration
+    - [x] Use Signal Expiration
     """
 
-    version = VersionProperty("0.0.1")
+    version = VersionProperty('0.1.0')
     min_signals = IntProperty(default=1, title='Min Signals')
     max_signals = IntProperty(default=20, title='Max Signals')
-    expiration = TimeDeltaProperty(title='Window Expiration',
-                                   allow_none=True)
+    expiration = TimeDeltaProperty(default={'seconds': -1},
+                                   title='Window Expiration',
+                                   advanced=True)
 
     def __init__(self):
         super().__init__()
         self._buffers = defaultdict(list)
-        self._last_recv = defaultdict(lambda : datetime.min)
+        self._last_recv = defaultdict(lambda : monotonic())
 
     def expire(self):
         self.logger.debug('Clearing the buffer window')
         self._buffers.clear()
 
     def process_group_signals(self, signals, group, input_id=None):
-        now = datetime.utcnow()
-
-        hasExpiration = self.expiration() is not None
-        if hasExpiration and (self._last_recv[group] + self.expiration()) < now:
+        hasExpiration = self.expiration().total_seconds() >= 0
+        isExpired = self._last_recv[group] > self.expiration().total_seconds()
```

Comment:
Does this actually expire? I don't see us recomputing `monotonic()` again, we need to compare it at some point. Am I missing something?